src/main/java/org/opentripplanner/netex/mapping/StopPlaceTypeMapper.java

```diff
@@ -12,73 +12,64 @@
  */
 class StopPlaceTypeMapper {
 
-  public T2<TransitMode, String> map(StopPlace stopPlace) {
-    T2<TransitMode, String> submode = getSubmodeAsString(stopPlace);
+  public NetexMainAndSubMode map(StopPlace stopPlace) {
+    NetexMainAndSubMode submode = getSubmodeAsString(stopPlace);
     if (submode != null) {
       return submode;
     }
     TransitMode mode = mapVehicleMode(stopPlace);
-    return new T2<>(mode, null);
+    return new NetexMainAndSubMode(mode, null);
   }
 
   private TransitMode mapVehicleMode(StopPlace stopPlace) {
     if (stopPlace.getTransportMode() == null) {
       return null;
     }
-    switch (stopPlace.getTransportMode()) {
-      case AIR:
-        return TransitMode.AIRPLANE;
-      case BUS:
-        return TransitMode.BUS;
-      case TROLLEY_BUS:
-        return TransitMode.TROLLEYBUS;
-      case CABLEWAY:
-        return TransitMode.CABLE_CAR;
-      case COACH:
-        return TransitMode.COACH;
-      case FUNICULAR:
-        return TransitMode.FUNICULAR;
-      case METRO:
-        return TransitMode.SUBWAY;
-      case RAIL:
-        return TransitMode.RAIL;
-      case TRAM:
-        return TransitMode.TRAM;
-      case WATER:
-      case FERRY:
-        return TransitMode.FERRY;
-      default:
-        return null;
-    }
+    return switch (stopPlace.getTransportMode()) {
+      case AIR -> TransitMode.AIRPLANE;
+      case BUS -> TransitMode.BUS;
+      case TROLLEY_BUS -> TransitMode.TROLLEYBUS;
+      case CABLEWAY -> TransitMode.CABLE_CAR;
+      case COACH -> TransitMode.COACH;
+      case FUNICULAR -> TransitMode.FUNICULAR;
+      case METRO -> TransitMode.SUBWAY;
+      case RAIL -> TransitMode.RAIL;
+      case TRAM -> TransitMode.TRAM;
+      case WATER, FERRY -> TransitMode.FERRY;
+      default -> null;
```

Comment:
Which modes are not handled here?

Reply:
I will change it to `case LIFT, OTHER, SNOW_AND_ICE -> null;` witch forces us to update this code in case a mode is added.

Reply:
We should probably throw an exception and not return null - but I do not want to make any functional changes in this PR.

Reply:
Thanks, we can do that in a separate PR